subprojects/ce-login/celogin/src/CeLoginJson.cpp

```diff
@@ -723,3 +724,182 @@ CeLoginRc ParseAdminResetFields(const JsmnUtils::JsmnState& jsmnStateParm,
+                                    sizeof(decodedJsonParm.mAsciiScriptFile),
+                                    decodedJsonParm.mAsciiScriptFileLength);
+    }
+
+    return sRc;
+}
+
+CeLoginRc ParseBmcShellFields(const JsmnUtils::JsmnState& jsmnStateParm,
+                              const AcfVersion versionParm,
+                              const char* serialNumberParm,
+                              const uint64_t serialNumberLengthParm,
+                              CeLoginJsonData& decodedJsonParm)
+{
+    CeLoginRc sRc = CeLoginRc::Success;
+    JsmnUtils::JsmnUtilRc sJsmnRc = JsmnUtils::Success;
+
+    const uint64_t sRootObjectTokenIdx = 0;
+
+    uint64_t sBmcShellIdx = 0;
+    uint64_t sBmcTimeoutIdx = 0;
+    uint64_t sIssueBmcDumpIdx = 0;
+
+    if (JsmnUtils::Success == sJsmnRc)
+    {
+        sJsmnRc = JsmnUtils::ObjectGetValueByKey(
+            jsmnStateParm, sRootObjectTokenIdx, JsonName_BmcShellScript,
+            strlen(JsonName_BmcShellScript), sBmcShellIdx);
+    }
+
+    if (JsmnUtils::Success == sJsmnRc)
+    {
+        sJsmnRc = JsmnUtils::ObjectGetValueByKey(
+            jsmnStateParm, sRootObjectTokenIdx, JsonName_BmcTimeoutVal,
+            strlen(JsonName_BmcTimeoutVal), sBmcTimeoutIdx);
+    }
+
+    if (JsmnUtils::Success == sJsmnRc)
+    {
+        sJsmnRc = JsmnUtils::ObjectGetValueByKey(
+            jsmnStateParm, sRootObjectTokenIdx, JsonName_IssueBmcDump,
+            strlen(JsonName_IssueBmcDump), sIssueBmcDumpIdx);
+    }
+
+    // Convert to CeLoginRc for consolidated RC handling
+    sRc = CeLoginRc(CeLoginRc::JsmnUtils, sJsmnRc);
+
+    if (CeLoginRc::Success == sRc)
+    {
+        ServiceAuthority sIgnoredAuth;
+        sRc = ParseCommonAcfFields(
+            jsmnStateParm, decodedJsonParm.mExpirationDate, serialNumberParm,
+            serialNumberLengthParm, sIgnoredAuth);
+    }
+
+    // Copy out bmcshell field
+    if (CeLoginRc::Success == sRc)
+    {
+        JsmnUtils::JsmnString sBmcShellStr =
+            jsmnStateParm.getString(sBmcShellIdx);
+
+        sRc = CeLogin::base64Decode(sBmcShellStr.mCharArray,
+                                    sBmcShellStr.mCharLength,
+                                    decodedJsonParm.mAsciiScriptFile,
+                                    sizeof(decodedJsonParm.mAsciiScriptFile),
+                                    decodedJsonParm.mAsciiScriptFileLength);
+    }
+
+    if (CeLoginRc::Success == sRc)
+    {
+        JsmnUtils::JsmnString sBmcTimeoutStr =
+            jsmnStateParm.getString(sBmcTimeoutIdx);
+
+        sRc = getUnsignedIntegerFromString(sBmcTimeoutStr.mCharArray,
+                                           sBmcTimeoutStr.mCharLength,
+                                           decodedJsonParm.mBmcTimeout);
+    }
+
+    if (CeLoginRc::Success == sRc)
+    {
+        JsmnUtils::JsmnString sIssueBmcDumpStr =
+            jsmnStateParm.getString(sIssueBmcDumpIdx);
```

Comment:
@bjvette-ibm with regards to the boolean concern, the issue is here. All types, regardless of whether they are a string, integer, bool, etc. end up needing to be parsed out from a string. So, if I used the boolean type, I'd need to investigate how to parse out the boolean definition from whatever string representation it puts it into/probably write some common parsing code, where as if I just use yes/no, it's just a simple memcmp.

Reply:
ack